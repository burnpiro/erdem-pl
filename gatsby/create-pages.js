'use strict';

const moment = require('moment');
const path = require('path');
const _ = require('lodash');
const createTagsPages = require('./pagination/create-tags-pages.js');
const createPostsPages = require('./pagination/create-posts-pages.js');

const createPages = async ({ graphql, actions }) => {
  const { createPage } = actions;

  // 404
  createPage({
    path: '/404',
    component: path.resolve('./src/templates/not-found-template.js')
  });

  // Tags list
  createPage({
    path: '/tags',
    component: path.resolve('./src/templates/tags-list-template.js')
  });

  // Posts and pages from markdown
  const results = await graphql(`
    {
      allBloggerPost(
        filter: { }
      ) {
        edges {
          node {
            fields {
              postSlug
            }
            slug,
            published
          }
        }
      }
      allBloggerPage(
        filter: { }
      ) {
        edges {
          node {
            slug,
            published
          }
        }
      }
    }
  `);

  const { allBloggerPost, allBloggerPage } = results.data;

  _.each(allBloggerPost.edges, (edge) => {
    createPage({
      path: edge.node.fields.postSlug,
      component: path.resolve('./src/templates/post-template.js'),
      context: { slug: edge.node.slug }
    });
  });

  _.each(allBloggerPage.edges, (edge) => {
    createPage({
      path: `/pages/${edge.node.slug}`,
      component: path.resolve('./src/templates/page-template.js'),
      context: { slug: edge.node.slug }
    });
  });

  // Feeds
  await createTagsPages(graphql, actions);
  await createPostsPages(graphql, actions);
};


module.exports = createPages;
