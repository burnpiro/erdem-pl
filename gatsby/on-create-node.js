const moment = require('moment');
const readingTime = require('reading-time');
const _ = require('lodash');
const { fmImagesToRelative } = require('gatsby-remark-relative-images');

const onCreateNode = ({ node, actions }) => {
  const { createNodeField } = actions;

  fmImagesToRelative(node);

  if (node.internal.type === 'MarkdownRemark') {
    if (typeof node.frontmatter.slug !== 'undefined') {
      createNodeField({
        node,
        name: 'slug',
        value: node.frontmatter.slug,
      });
    } else {
      createNodeField({
        node,
        name: 'slug',
        value: `/${moment(node.date).format('YYYY/MM')}/${_.kebabCase(
          node.frontmatter.title
        )}`,
      });
    }

    if (node.frontmatter.tags) {
      const tagSlugs = node.frontmatter.tags.map(
        tag => `/tag/${_.kebabCase(tag)}/`
      );
      createNodeField({ node, name: 'tagSlugs', value: tagSlugs });
    }

    if (node.frontmatter.template === 'post') {
      const readTime = readingTime(node.internal.content);
      createNodeField({ node, name: 'readTime', value: readTime });
    }
  }
};

module.exports = onCreateNode;
