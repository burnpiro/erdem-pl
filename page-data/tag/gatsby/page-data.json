{"componentChunkName":"component---src-templates-tag-template-js","path":"/tag/gatsby","webpackCompilationHash":"37ec7ece8b5c4a2c9a8b","result":{"data":{"site":{"siteMetadata":{"title":"Blog by Kemal Erdem","subtitle":"Even complex things could be explain in a simple way."}},"allMarkdownRemark":{"edges":[{"node":{"fields":{"readTime":{"text":"13 min read","minutes":12.47},"slug":"/2017/08/johannes-gutenberg-the-birth-of-movable-type"},"frontmatter":{"date":"2017-08-18","title":"Johannes Gutenberg: The Birth of Movable Type","description":"German inventor Johannes Gutenberg developed a method of movable type and used it to create one of the western world’s first major printed books, the “Forty–Two–Line” Bible."}}}]}},"pageContext":{"isCreatedByStatefulCreatePages":false,"tag":"Gatsby","currentPage":0,"postsLimit":4,"postsOffset":0,"prevPagePath":"/tag/gatsby","nextPagePath":"/tag/gatsby/page/1","hasPrevPage":false,"hasNextPage":false}}}