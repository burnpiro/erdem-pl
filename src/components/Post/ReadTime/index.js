// @flow
export { default } from './ReadTime';
