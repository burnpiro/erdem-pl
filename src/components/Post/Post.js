// @flow
import React from 'react';
import { Link } from 'gatsby';
import Author from './Author';
import Comments from './Comments';
import Content from './Content';
import Meta from './Meta';
import Tags from './Tags';
import styles from './Post.module.scss';
import type { BloggerPost } from '../../types';

type Props = {
  post: BloggerPost,
};

const Post = ({ post }: Props) => {
  const { content } = post;
  const {
    slug,
    fields: { tagSlugs, readTime },
  } = post;
  const { labels, title, published } = post;

  return (
    <div className={styles['post']}>
      <Link className={styles['post__home-button']} to="/">
        All Articles
      </Link>

      <div className={styles['post__content']}>
        <Content
          body={content}
          title={title}
          date={published}
          readTime={readTime}
        />
      </div>

      <div className={styles['post__footer']}>
        <Meta date={published} />
        {labels && tagSlugs && <Tags tags={labels} tagSlugs={tagSlugs} />}
        <Author />
      </div>

      <div className={styles['post__comments']}>
        <Comments postSlug={slug} postTitle={post.title} />
      </div>
    </div>
  );
};

export default Post;
