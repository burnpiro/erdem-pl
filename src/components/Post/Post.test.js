// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Post from './Post';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import markdownRemark from '../../../jest/__fixtures__/markdown-remark';
import type { RenderCallback } from '../../types';

describe('Post', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => render(siteMetadata),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  const props = {
    post: {
      id: 'test-123',
      html: '<p>test</p>',
      htmlAst: { children: [], data: { quirksMode: false }, type: 'root' },
      fields: {
        slug: '/2020/02/understanding-region-of-interest-ro-i-pooling',
        categorySlug: '/test-category',
        tagSlugs: ['/test_0', '/test_1'],
      },
      frontmatter: {
        date: '2016-09-01',
        tags: ['test_0', 'test_1'],
        title: 'test',
      },
    },
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Post {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
