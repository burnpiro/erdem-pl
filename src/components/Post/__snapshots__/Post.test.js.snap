// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders correctly 1`] = `
<div
  className="post"
>
  <a
    className="post__home-button"
    href="/"
  >
    All Articles
  </a>
  <div
    className="post__content"
  >
    <div
      className="content"
    >
      <h1
        className="content__title"
      >
        test
      </h1>
      <div
        className="meta"
      >
        <span
          className="meta__date"
        >
          Published 
          1 Sep 2016
        </span>
      </div>
      <div
        className="content__body"
      >
        <div />
      </div>
      <div
        className="citation"
      >
        <h3>
          Citation
        </h3>
        <blockquote
          className="citation__text"
        >
          Test name, (Sep 2016). "test". 
          <a
            href="http://localhost/2020/02/understanding-region-of-interest-ro-i-pooling"
          >
            http://localhost/2020/02/understanding-region-of-interest-ro-i-pooling
          </a>
        </blockquote>
        <b>
          or
        </b>
        <pre
          className="citation__code"
          data-language="javascript"
        >
          <code>
            @article{erdem2016understandingRegionOfInterestRoIPooling,
    title   = "test",
    author  = "Test name",
    journal = "http://localhost",
    year    = "2016",
    month   = "Sep",
    url     = "http://localhost/2020/02/understanding-region-of-interest-ro-i-pooling"
}

          </code>
        </pre>
      </div>
    </div>
  </div>
  <div
    className="post__footer"
  >
    <div
      className="meta"
    >
      <span
        className="meta__date"
      >
        Published 
        1 Sep 2016
      </span>
    </div>
    <div
      className="tags"
    >
      <ul
        className="tags__list"
      >
        <li
          className="tags__list-item"
        >
          <a
            className="tags__list-item-link"
            href="/test_0"
          >
            test_0
          </a>
        </li>
        <li
          className="tags__list-item"
        >
          <a
            className="tags__list-item-link"
            href="/test_1"
          >
            test_1
          </a>
        </li>
      </ul>
    </div>
    <div
      className="author"
    >
      <p
        className="author__bio"
      >
        Test bio
        <a
          className="author__bio-twitter"
          href="https://www.twitter.com/#"
          rel="noopener noreferrer"
          target="_blank"
        >
          <strong>
            Test name
          </strong>
           on Twitter
        </a>
      </p>
    </div>
  </div>
  <div
    className="post__comments"
  />
</div>
`;
